supervisor/hyperpod.go

```diff
@@ -366,7 +366,12 @@ func (hp *HyperPod) startNsListener() (err error) {
 
 func (hp *HyperPod) stopNsListener() {
 	if hp.nslistener != nil {
-		hp.nslistener.cmd.Process.Kill()
+		go func() {
+			hp.nslistener.cmd.Process.Kill()
+		}()
+		if err := hp.nslistener.cmd.Process.Signal(syscall.Signal(0)); err == nil {
+			hp.nslistener.cmd.Wait()
```

Comment:
use time.AfterFunc()

"if hp.nslistener != nil" ensures the process is/was launched